source/retrieve-data.txt

```diff
@@ -0,0 +1,196 @@
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:ref:`<java-rs-getting-started>`.
+
+.. include:: includes/reactor-note.rst
+
+.. _java-rs-retrieve-find:
+
+Find Documents
+--------------
+
+The {+driver-short+} includes one method for retrieving documents from a collection:
+``find()``.
+
+This method takes a **query filter** and returns one or more matching documents.
+A query filter is an object that specifies the documents you want to retrieve in
+your query. 
+
+.. TODO: uncomment when specify query is merged in
+.. To learn more about query filters, see :ref:`java-rs-specify-query`.
+
+.. _java-rs-retrieve-find-one:
+
+Find One Document
+~~~~~~~~~~~~~~~~~
+
+To find a single document in a collection, call the ``find()`` method and pass a query
+filter that specifies the criteria of the document you want to find. Then, chain
+the ``first()`` method to find a single document. If more than one document matches the query
+filter, this construct returns the *first* matching document from the retrieved
+results. If no documents match the query filter, the method returns
+``None``.
+
+.. tip::
+   
+   The ``find().first()`` construct is useful when you know there's only one matching document,
+   or you're only interested in the first match.  
+
+The following example uses the ``find().first()`` construct to find the first document where
+the ``"cuisine"`` field has the value ``"Bakery"``:
+
+.. code-block:: java
+   :copyable: true
+
+   Publisher<Document> findDocPublisher = restaurants.find(
+       eq("cuisine", "Bakery")).first();
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+   
+.. tip:: Sort Order
+
+   The ``find().first()`` construct returns the first document in 
+   :manual:`natural order </reference/glossary/#std-term-natural-order>`
+   on disk if no sort criteria is specified.
+
+.. TODO: uncomment when sort guide made
+.. To learn more about sorting, see the :ref:`sort guide <java-rs-sort>`.
+
+.. _java-rs-retrieve-find-multiple:
+
+Find Multiple Documents
+~~~~~~~~~~~~~~~~~~~~~~~
+
+To find multiple documents in a collection, pass a query filter to the ``find()``
+method that specifies the criteria of the documents you want to retrieve.
+
+The following example uses the ``find()`` method to find all documents where
+the ``"cuisine"`` field has the value ``"Spanish"``:
+
+.. code-block:: java
+   :copyable: true
+
+   FindPublisher<Document> findDocPublisher = restaurants.find(
+           eq("cuisine", "Spanish"));
+   List<Document> findResult = Flux.from(findDocPublisher).collectList().block();
+
+You can iterate over the documents you retrieved using the ``find()`` method by using a ``for`` loop, as shown in
+the following example:
```

Comment:
S: for length and clarity. probably no need to specify how the document were retrieved

```suggestion
You can iterate over the found documents by using a ``for`` loop, as shown in
the following example:
```